### DISCUSSION TITLE
HttpResponse Code canceled and still being billed

### DISCUSSION THREAD

**Discussion by bmaluijb** [deepgram/community#277] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Hi,

I just called the transcription API for a long audio file (3,5 hours), and get HTTP Code Canceled back from the API. I use the C# library Polly to automatically retry on an error like this, which works. However, in my log, I see that I do have usage from this call and from the subsequent calls that result in a canceled response, which results in me paying 5 times for one successful call.

Can you take a look at what's happening please?

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?tier=nova&punctuate=true&diarize=true&ner=false&smart_format=true&utterances=true&paragraphs=true&detect_entities=false

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

eb9c5ac6-5330-48c1-99c5-c0523336889e

### If possible, please attach your code or paste it into the text box.

Also for request ids:
3c15c85d-7616-4165-b082-061b9288b4f0
43488e50-632d-4dc7-a650-fcbfe2660cd2
b13fc78a-4c9d-4467-95d8-47b086af8d36

It finally worked for request ID 
25adb4df-0159-402a-81ad-c87d0c7150d9

C# Code:
            var retryPolicy =
                Policy.Handle<Exception>()
                //retry 5 times with exponential backoff
                .WaitAndRetryAsync(5, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)));

            response = await retryPolicy.ExecuteAsync(async () =>
            {
                return await deepgram.Transcription.Prerecorded.GetTranscriptionAsync(
                new UrlSource(episode.TemporaryAudioURL), options);
            });

### If possible, please attach an example audio file to reproduce the issue.

episode.TemporaryAudioURL = https://podhomemediastorage.blob.core.windows.net/media/6382529988713508755b6c2e27-9d77-4718-9936-f1827570e806.mp3

**Comment by jkroll-deepgram** [comment]
Hi @bmaluijb, can you share more about the exact error message and status code you are receiving? I see that each of these requests took about 2 minutes to process, and finished processing the request internally, but then only the successful one returned a response afterward. Meanwhile I will look further into when and how the responses were dropped.

**Comment by bmaluijb** [comment]
Hi Julia,

These were HTTP 400 responses, and had a description of “Cancelled”. On my side however, these requests only ran for about 30 seconds, which would not cause my side to time-out. I don’t have any other info than that at the moment.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jkroll-deepgram** [comment]
Hi @bmaluijb, on further investigation, Deepgram did eventually return responses and our side shows 200s. Can you elaborate on where you were seeing the "cancelled" status? I confirmed that we don't use that particular error code. I wonder if the issue is in waiting long enough for Deepgram's response, so that after ~2 minutes, you can still receive the result?

**Comment by bmaluijb** [comment]
Canceled is the Http status code that, I guess, my Azure Application Insights gives to this particular issue. I have no further info for now. So let’s consider this an anomaly and I’ll keep an eye out to see if this occurs more.

Thank you for continuing to investigate this!

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by bmaluijb** [comment, marked as answer]
I think I found the issue. 

I'm using the .NET DeepGram client from NuGet, like this:

            var deepgram = new DeepgramClient(credentials);

It seems to use the default HttpClient Timeout of 100 seconds, and fails quietly when it times out. I've now used:

            deepgram.SetHttpClientTimeout(TimeSpan.FromMinutes(30));

I need to thoroughly test t, but I think this is it. Perhaps you can increase the default timeout by default in the SDK.
